packages/core/src/constant.ts

```diff
@@ -10,3 +10,41 @@ export const DEFAULT_EXTENSIONS = [
 ] as const;
 
 export const SWC_HELPERS = '@swc/helpers';
+
+export const JS_EXTENSIONS: string[] = [
+  'js',
+  'mjs',
+  'jsx',
+  'ts',
+  'mts',
+  'tsx',
+  'cjs',
+  'cjsx',
+  'mjsx',
+  'mtsx',
+  'cts',
+  'ctsx',
+] as const;
+
+export const CSS_EXTENSIONS: string[] = [
+  'css',
+  'sass',
+  'scss',
+  'less',
+] as const;
+
+export const JSON_EXTENSIONS: string[] = ['json'] as const;
+
+export const ENTRY_EXTENSIONS: string[] = [
+  ...JS_EXTENSIONS,
+  ...CSS_EXTENSIONS,
+  ...JSON_EXTENSIONS,
```

Comment:
json is aligned with modernjs module, no corresponding tests in its repo yet. Json will be treated as js as same?

Reply:
Json is like an image.

- It could be imported and preserve as-is by other JS files
- It could be inlined and tree shaken.
- It could not import other modules.

So since image won’t be used as an entry, shouldn’t json be used as an entry either?

Reply:
You are right, we should remove json from entries too.